### PR TITLE
fix: excessive memory usage

### PR DESCRIPTION
**Pull request by peusebiu**
instead of reading entire files before calculating their digests stream them by using their Reader method.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/scrub.go [hunk 1]
```diff
@@ -279,21 +279,28 @@ func CheckLayers(
...
+			break
+		}
 
 		if computedDigest != layer.Digest {
 			imageRes = getResult(imageName, tagName, layer.Digest, errors.ErrBadBlobDigest)
 
 			break
 		}
 
+		_ = layerReader.Close()
```
Comment (andaaron) [inline pkg/storage/scrub.go:302]: We need to make sure this is called, even on the `break` cases above.
Why not move the entire code block to a separate function?

**Comment by rchincha** [comment]
@peusebiu can we see the effect of this change when a image with several large blobs need to be scrubbed?

**Comment by peusebiu** [comment]
> @peusebiu can we see the effect of this change when a image with several large blobs need to be scrubbed?

Ok I needed to handle a lock case, so I made an update to add a new storage method which doesn't lock.

Yes tested it and no memory spike at all, on my machine is stays on a steady 1.3% memory, on main it has a lot of spikes when it's digesting big blobs.

File: pkg/storage/types/types.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ type ImageStore interface { //nolint:interfacebloat
 	PutIndexContent(repo string, index ispec.Index) error
 	StatIndex(repo string) (bool, int64, time.Time, error)
 	GetBlobContent(repo string, digest godigest.Digest) ([]byte, error)
+	GetBlobReader(repo string, digest godigest.Digest) (io.ReadCloser, int64, error)
```
Comment (andaaron) [inline pkg/storage/types/types.go:60]: It is not clear how this is different from the method at line 51

```suggestion
        VerifyBlobDigestValue(repo string, digest godigest.Digest) error
```

Let's call this function from the scrub logic.

**Pull request merged**